test/Catch.hs

```diff
@@ -12,20 +12,20 @@ import Test.Tasty.Hedgehog
 genN
   :: forall e m a sig
   .  (Has (Catch e) sig m, Arg e, Show e, Vary e)
-  => Gen e
+  => GenTerm e
   -> GenM m
-  -> Gen a
-  -> [Gen (m a)]
-genN _ m a = [ label "catchError" catchError <*> m a <*> fn @e (m a) ]
+  -> GenTerm a
+  -> [GenTerm (m a)]
+genN _ m a = [ addLabel "catchError" $ subtermM (m a) (\ m' -> infixL 9 "`catchError`" catchError <*> m' <*> fn @e (m a)) ]
```

Comment:
Two things worth noting here:

1. We’re generating `catchError` as an infix term, so it’ll actually be shown as:

    ```haskell
    m `catchError` h
    ```

2. We use `subtermM` to generate the embedded action (but not the handler, because that would limit the distribution unnecessarily), which allows it to shrink to that action.